### ISSUE TITLE
Spine experiencing MySQL socket error 2002 under load

### ISSUE THREAD

**Issue by whsir** [Cacti/spine#171]
SPINE: Using spine config file [spine.conf]
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)'
2020/09/15 17:28:09 - SPINE: Poller[1] PID[13318] FATAL: Connection Failed, Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)' (Spine init)

my mysql sock /tmp/mysql.sock
so
I can only 'ln -sv /tmp/mysql.sock /var/lib/mysql/mysql.sock'

**Comment by TheWitness** [comment]
What is in your my.cnf | server.cnf | client.cnf and your spine.conf?

**Comment by whsir** [comment]
my.conf  
socket = /tmp/mysql.sock

**Comment by whsir** [comment]
run /usr/local/spine/bin/spine
SPINE: Poller[1] PID[13318] FATAL: Connection Failed, Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (2)' (Spine init)
so
I can only 'ln -sv /tmp/mysql.sock /var/lib/mysql/mysql.sock'

**Comment by TheWitness** [comment]
There is a setting in spine to point it to the socket.  Can't remember the setting though.

**Comment by TheWitness** [comment]
You can set DB_HOST to /tmp/mysql.sock in spine.conf

**Comment by whsir** [comment]
OK
spine.conf
DB_Host       /tmp/mysql.sock
That should make it look normal！
Doesn't that seem a little unreasonable？

**Comment by TheWitness** [comment]
Well, I think if you want to keep using the socket, you should add the option to the client section of the MySQL configuration.  In modern mariadb's that would be /etc/my.cnf.d/client.cnf. That way you can continue to use localhost.

**Comment by whsir** [comment]
but，This was not the case in previous versions。

For example, SPINE 1.2.2 version

**Comment by TheWitness** [comment]
I don't think it has anything to do with spine.

**Comment by bmfmancini** [comment]
I have the same issue  but in my case the errors creep up 
I am on spine 1.2.12 in prod and see these errors come in batches 

This is a sample of my errors 

```
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
MYSQL: Connection Failed: Error:'2002', Message:'Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (11 "Resource temporarily unavailable")'
```

Not exceeding connections 
No memory level issue
Proccess has been up for a while

**Comment by TheWitness** [comment]
Been seeing so many issues with MariaDB lately with 2006 and 2013 errors, and these 2002 errors just add to the intrigue.

**Comment by TheWitness** [comment]
Can you guys follow the recommendations here?  Provide feedback too.

https://stackoverflow.com/questions/10131615/mysql-resource-temporarily-unavailable

I also made some updates, but not for the 2002 errors.

**Comment by TheWitness** [comment]
I've made an update to spine.  Please test.

**Comment by TheWitness** [comment]
I'm going to assume this is resolved now.  I've implemented connection pooling with #187.  So, that should help.
